osgar/drivers/test_gps.py

```diff
@@ -11,6 +11,9 @@ def test_checksum(self):
     def test_str2ms(self):
         self.assertEqual(gps.str2ms(b'5007.71882'), 180463129)
 
+    def test_str2ms_err_input(self):
+        self.assertEqual(gps.str2ms(b'01422.489915\x0052'), None)
```

Comment:
could you add here note/comment, that the checksum is passing because it adds 0x00? It should fail that there are non-ASCII characters

Reply:
Ok

Reply:
> Thanks for explanation - that looks a bit scary :( ... is there some other source inserting zero bytes into the GPS stream? I noticed that checksum failure in this very old code means module termination ... which is not very good, if the line is not reliable :(. I.e. have you seen also other extra bytes?

I don't know. This data were recorded using a rs232/usb converter so probably this is the source. I have checked some other logs and i did not notice this behavior. However, that logs were recorded without the rs232/usb converter.